packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -165,16 +180,29 @@ export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvi
             );
             this.dataProvider.onContentChanged(elementIds);
         }
+
         if (event.eventType == PluginEventType.KeyDown) {
-            this.eventHandledOnKeyDown = false;
-            this.onKeyDownEvent(event);
+            if (event.rawEvent.key == UNIDENTIFIED_KEY) {
+                // On Android, the key for KeyboardEvent is "Unidentified",
+                // so ignore onKeyDownEvent and handle with InputEvent instead
+                this.isPendingInputEventHandling = true;

```

Comment:
Instead of calculating currentInputLength when insertNode, I think it will be more accurate to calculate the content length here, because here means input event is about to happen, so we can get content length before the input. Please work with John to see if it works correctly.

Reply:
Sounds good. I've updated accordingly and asked John to see if the new code still works correctly 👍🏻